src/invocation/ClusterService.ts

```diff
@@ -143,7 +143,8 @@ class ClusterService extends EventEmitter {
                     deferred.reject(error);
                     return;
                 } else {
-                    setTimeout(this.tryAddressIndex(0, attemptLimit, attemptPeriod, deferred), attemptPeriod);
+                    var self = this;
+                    setTimeout(function() {self.tryAddressIndex(0, attemptLimit, attemptPeriod, deferred)}, attemptPeriod);
```

Comment:
Can you change the function decleration to a function binding as `this.tryAddressIndex.bind(this, 0, attemptLimit, attemptPeriod, deferred)`. It is just to be consistent with the rest of the codebase. Then it's ok. Thanks for the fix.